src/init/common.cpp

```diff
@@ -80,26 +79,31 @@ util::Result<void> SetLoggingLevel(const ArgsManager& args)
     return {};
 }
 
-util::Result<void> SetLoggingCategories(const ArgsManager& args)
+[[nodiscard]] static bool EnableOrDisable(const std::string& opt, const std::string& category)
 {
-    if (args.IsArgSet("-debug")) {
-        // Special-case: if -debug=0/-nodebug is set, turn off debugging messages
-        const std::vector<std::string> categories = args.GetArgs("-debug");
-
-        if (std::none_of(categories.begin(), categories.end(),
-            [](std::string cat){return cat == "0" || cat == "none";})) {
-            for (const auto& cat : categories) {
-                if (!LogInstance().EnableCategory(cat)) {
-                    return util::Error{strprintf(_("Unsupported logging category %s=%s."), "-debug", cat)};
-                }
-            }
-        }
+    if (opt == "-debug") return LogInstance().EnableCategory(category);
+    if (opt == "-debugexclude") return LogInstance().DisableCategory(category);
+    NONFATAL_UNREACHABLE();
+}
+
+// Maintain similar logic in both of the EnableOrDisableLogCategories() functions in the codebase.
```

Comment:
I still don't totally understand why you need to split this up into three functions, one of which is a kinda-duplicate The comment makes sense but I'm not familiar enough with the general style of the project to determine if having two functions with the same name doing basically the same thing is acceptable.

Reply:
OK cool thanks for explaining